public/js/components/Dropdown.js

```diff
@@ -0,0 +1,63 @@
+const React = require("react");
+const { DOM: dom, PropTypes } = React;
+
+const Dropdown = React.createClass({
+  propTypes: {
+    panel: PropTypes.object
+  },
+
+  displayName: "Dropdown",
+
+  getInitialState() {
+    return {
+      dropdownShown: false
+    };
+  },
+
+  toggleSourcesDropdown(e) {
+    this.setState({
+      dropdownShown: !this.state.dropdownShown,
+    });
+  },
+
+  renderPanel() {
+    return dom.div(
+      {
+        className: "sources-dropdown dropdown",
```

Comment:
Please don't use the term "sources" in a generic that is not tied to sources at all. If this is supposed to be a generic dropdown component, just use generic terms.